dataset/src/main/scala/frameless/functions/Udf.scala

```diff
@@ -0,0 +1,80 @@
+package frameless.functions
+
+import frameless.{TypedColumn, TypedEncoder}
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions.{udf => sudf}
+
+import scala.reflect.runtime.universe._
+
+trait Udf {
+  def udf[T, A: TypeTag, R: TypeTag](
+     f: A => R)
+  (implicit
+     aenc: TypedEncoder[A],
+     reng: TypedEncoder[R]):
+  (TypedColumn[T, A] => TypedColumn[T, R]) =
+    (u: TypedColumn[T, A]) => {
+      val uf = sudf(f)
```

Comment:
It shows one of the problems with our tests, probably we should also test against something that is only encodable by frameless :)